test/plugin/base_test.rb

```diff
@@ -536,6 +536,35 @@ def test_structured_payload_json_log_detect_json_not_parsed_json
     end
   end
 
+  def test_structured_payload_json_log_detect_json_not_parsed_hash
+    hash_value = {
+      'msg' => 'test log entry 0',
+      'tag2' => 'test',
+      'data' => 5000,
+      'some_null_field' => nil
+    }
+    %w(log msg).each do |field|
```

Comment:
Why not also `message`?

Reply:
`message` behaves differently in general (which is why multiple other tests do `log msg` only as well: https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/367427e4cb0b619e27b9c5b77f7080eeff4cb4eb/lib/fluent/plugin/out_google_cloud.rb#L2034.

Feels weird, but we couldn't easily change it (maybe the nest major release).

Reply:
Right, but the JSON detection code runs on all three: https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/367427e4cb0b619e27b9c5b77f7080eeff4cb4eb/lib/fluent/plugin/out_google_cloud.rb#L640.
What happens if we pass a Hash in the `message` field?

Reply:
Right, but the JSON detection code runs on all three: https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/367427e4cb0b619e27b9c5b77f7080eeff4cb4eb/lib/fluent/plugin/out_google_cloud.rb#L640.
What happens if we pass a Hash in the `message` field?

Reply:
Added a test and a TODO to fix the inconsistency.